libs/schema-profunctor/src/Data/Schema.hs

```diff
@@ -563,6 +563,8 @@ instance ToSchema String where schema = genericToSchema
 
 instance ToSchema Bool where schema = genericToSchema
 
+instance (S.ToSchema a, A.ToJSON a, A.FromJSON a)=> ToSchema [a] where schema = genericToSchema
```

Comment:
I would prefer this:

```suggestion
instance ToSchema a => ToSchema [a] where schema = array schema
```

Reply:
Hmm, is there a difference. This does look better, asking for my education.
I also pushed another instance for `NonEmpty a` I am guessing `array` wouldn't produce a nice ToSchema instance for that, so we should use `genericToSchema` in that case?

Reply:
Hmm, `array schema` doesn't work as it doesn't produce a named schema, so either I need a `TypeRep`, which would be annoying or I need to set it to something like an empty string.

Reply:
Ah, you're right, on both counts.

I think it's better not to have this instance. One can just use `array` when you need an array, since lists are indeed most commonly unnamed.

Similarly, I would add a `nonEmptyArray` combinator instead of the `NonEmpty` instance.